src/hotspot/share/opto/vtransform.cpp

```diff
@@ -40,36 +40,68 @@ void VTransformGraph::add_vtnode(VTransformNode* vtnode) {
     }                                                 \
   )
 
-// This is similar to IGVN optimization. But we are a bit lazy, and don't care about
-// notification / worklist, since the list of nodes is rather small, and we don't
-// expect optimizations that trickle over the whole graph.
```

Comment:
This was wrong: we did have have a "trickle" over the graph with a chain of dead nodes, progressively realizing that the last node in the link was dead.